File: WebContent/org/r3/db/cms/card/CardDTO.js
//
// GPA Gruppo Progetti Avanzati s.r.l. - 
// Progetto UNEP 2012 per PosteItaliane S.p.a.
//
//
// TODO: Readme Section.
//


jQuery.r3Utils.namespace("org.r3.db.cms.card");

org.r3.db.cms.card.CardDTO = Class.extend(
{
	 init: function(aDataBean)
	 {  
	 	 	 this.dtrowclass   = "grade_card";
       this.dtokey       = aDataBean.dtokey   || null;
       this.dtodescr     = aDataBean.dtodescr || null;
       
        this.site = aDataBean.site || null;
        this.cardid = aDataBean.cardid || null;
        this.systemtitle = aDataBean.systemtitle || null;
        this.pagelinktype = aDataBean.pagelinktype || null;
        this.pagelink = aDataBean.pagelink || null;
        this.pageid = aDataBean.pageid || null;
        this.refdate = aDataBean.refdate || null;
        this.options = aDataBean.options || null;
        this.validityrule = aDataBean.validityrule || null;
        this.validfrom = aDataBean.validfrom || null;
        this.validto = aDataBean.validto || null;
        this.recstatus = aDataBean.recstatus || null;
        this.rectimestamp = aDataBean.rectimestamp || null;
        this.categoryid = aDataBean.categoryid || null;
 
        // Inserito
        if ( aDataBean.groups)
        {
        	this.groups =  new org.r3.db.DTOTableAdapter(aDataBean.groups, 
    				function(aRowDataBean) { return new org.r3.db.cms.group.GroupDTOAdapter(aRowDataBean); });            	
        }        
        else this.groups = null;

        if ( aDataBean.channels)
        {
        	this.channels =  new org.r3.db.DTOTableAdapter(aDataBean.channels, 
    				function(aRowDataBean) { return new org.r3.db.cms.channel.ChannelDTOAdapter(aRowDataBean); });            	
        }        
        else this.channels = null;            
	 }
    
  ,compareTo: function(anItemId)
   {
   	   if (!this.dtokey)
   	       return -99;
   	       
   	   if (this.dtokey < anItemId)
   	       return -1;
   	   else if (this.dtokey > anItemId)
   	   	   return  1;
   	   	   
   	   return 0;
   }
   
});

File: src/org/r3/db/cms/card/CardDTO.java
//
// GPA Gruppo Progetti Avanzati s.r.l. - 
// Progetto UNEP 2012 per PosteItaliane S.p.a.
//
//
// TODO: Readme Section.
//


package org.r3.db.cms.card;

import org.r3.db.cms.channel.ChannelDataTableAdapter;
import org.r3.db.cms.group.GroupDataTableAdapter;

public class CardDTO extends CardDTOBase
{
	public enum ValidityRule { always, never, date };
	public enum PageLinkType { none, page, custom };
	
	public static final String CARDID_DOMAIN_SEQUENCE_CODE = "CRDD";
		
    public CardDTO()
    {
        super();
    }

    public CardDTO(CardDTO another)
    {
        super(another);
    }

    public static ValidityRule getValidityruleAsEnum(String v, ValidityRule defaultValue)
    {
    	if (v != null)
    	{
    		try
    		{
        		return ValidityRule.valueOf(v);    			
    		}
    		catch(Exception e)
    		{
    			e.printStackTrace();
    		}
    	}
    	
    	return defaultValue;
    }

    public static PageLinkType getPagelinktypeAsEnum(String v, PageLinkType defaultValue)
    {
    	if (v != null)
    	{
    		try
    		{
        		return PageLinkType.valueOf(v);    			
    		}
    		catch(Exception e)
    		{
    			e.printStackTrace();
    		}
    	}
    	
    	return defaultValue;
    }

    public ValidityRule getValidityruleAsEnum()
    {
    	return getValidityruleAsEnum(getValidityrule(), ValidityRule.never);
    }
    
    public PageLinkType getPagelinktypeAsEnum()
    {
    	return getPagelinktypeAsEnum(getPagelinktype(), PageLinkType.none);
    }
    
    private GroupDataTableAdapter groups;

	public GroupDataTableAdapter getGroups()
	{
		return groups;
	}

	public void setGroups(GroupDataTableAdapter groups)
	{
		this.groups = groups;
	}
    
    private ChannelDataTableAdapter channels;

	public ChannelDataTableAdapter getChannels()
	{
		return channels;
	}

	public void setChannels(ChannelDataTableAdapter channels)
	{
		this.channels = channels;
	}
	
	public static boolean isDomainScope(String aCardId)
	{
		if (aCardId != null && aCardId.startsWith(CARDID_DOMAIN_SEQUENCE_CODE))
			return true;
		
		return false;
	}
	
	public boolean isDomainScope()
	{	
		return isDomainScope(getCardid());
	}
}
